infrastructure/src/main/java/org/corfudb/infrastructure/CompactorLeaderServices.java

```diff
@@ -0,0 +1,497 @@
+            }
+            long totalTimeElapsed = System.currentTimeMillis() - managerStatus.getTimeTaken();
+            syslog.info("Total time taken for the compaction cycle: {}ms for {} tables", totalTimeElapsed,
+                    tableNames.size());
+            MicroMeterUtils.time(Duration.ofMillis(totalTimeElapsed), "compaction.total.timer",
+                    "nodeEndpoint", nodeEndpoint);
+            TransactionalContext.getCurrentContext().setPriorityLevel(CorfuMessage.PriorityLevel.HIGH);
+            txn.putRecord(compactionManagerTable, DistributedCompactor.COMPACTION_MANAGER_KEY,
+                    buildCheckpointStatus(cpFailed ? StatusType.FAILED : StatusType.COMPLETED,
+                            tableNames.size(), totalTimeElapsed),
+                    null);
+            txn.commit();
+        } catch (Exception e) {
+            syslog.warn("Exception in finishCompactionCycle: {}. StackTrace={}", e, e.getStackTrace());
+        }
+
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            RpcCommon.TokenMsg upgradeToken = (RpcCommon.TokenMsg) txn.getRecord(DistributedCompactor.CHECKPOINT,
+                    DistributedCompactor.UPGRADE_KEY).getPayload();
+            txn.delete(DistributedCompactor.CHECKPOINT, DistributedCompactor.UPGRADE_KEY);
+            txn.commit();
+            if (upgradeToken != null) {
+                syslog.info("Upgrade Key found: Hence invoking trimlog()");
+                trimLog();
+            }
+        }
+
+        syslog.info("Finished the compaction cycle");
+    }
+
+    /**
+     * Perform log-trimming on CorfuDB by selecting the smallest value from checkpoint map.
+     */
+    @VisibleForTesting
+    public void trimLog() {
+        RpcCommon.TokenMsg thisTrimToken;
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            CheckpointingStatus managerStatus = (CheckpointingStatus) txn.getRecord(
+                    DistributedCompactor.COMPACTION_MANAGER_TABLE_NAME,
+                    DistributedCompactor.COMPACTION_MANAGER_KEY).getPayload();
+            if (managerStatus.getStatus() == StatusType.COMPLETED) {
+                thisTrimToken = txn.getRecord(checkpointTable, DistributedCompactor.CHECKPOINT_KEY).getPayload();
+            } else {
+                syslog.warn("Skip trimming since last checkpointing cycle did not complete successfully");
+                return;
+            }
+            txn.commit();
+        } catch (Exception e) {
+            syslog.warn("Unable to acquire the trim token");
+            return;
+        }
+
+        if (thisTrimToken == null) {
+            syslog.warn("Trim token is not present... skipping.");
+            return;
+        }
+
+        // Measure time spent on trimming.
+        final long startTime = System.nanoTime();
+        corfuRuntime.getAddressSpaceView().prefixTrim(
+                new Token(thisTrimToken.getEpoch(), thisTrimToken.getSequence()));
```

Comment:
This should be a high priority request. Otherwise if the quota is full, this prefix trim will be rejected. This is also a nice scenario to test.

Reply:
Actually, this is already the case. What I had observed was an artifact of the runtime used in the tests.